src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -483,6 +507,23 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     isQuoteStale,
   });
 
+  /** Failed Quote analytics */
+  useAnimatedReaction(
+    () => quote.value,
+    quote => {
+      const inputAsset = internalSelectedInputAsset.value;
+      const outputAsset = internalSelectedOutputAsset.value;
+      if (quote && 'error' in quote && inputAsset && outputAsset) {
+        runOnJS(trackQuoteFailed)(quote, {
+          inputAsset,
+          outputAsset,
+          inputAmount: SwapInputController.inputValues.value.inputAmount,
+          outputAmount: SwapInputController.inputValues.value.outputAmount,
+        });
+      }
+    }
+  );
+
```

Comment:
Do we want to track the same failure data more than once? Currently our quote fetching is done on an interval so this would track each failure even if the input/output assets haven't changed since last failure.

Reply:
Why can't we handle this in `setQuote`?

Reply:
I think you could move that logic to a `runOnUI` call in here (so we can read the SVs on the UI thread):

```ts
if (!data || (data as QuoteError)?.error) {
    resetFetchingStatus({ fromError: true, quoteFetchingInterval });
    return;
}
```

(Maybe need to adjust that check slightly)

Reply:
I'm not sure I understood correctly is this it? https://github.com/rainbow-me/rainbow/pull/6287/files#diff-9813ac25397f1586b8f178c8a741139a4ad4f9bb105b48de61407530c4570945R41-R357

Reply:
I'd move the type/function/lastParams out of the useSwapInputsController file, otherwise looks good